chain/sync.go

```diff
@@ -379,7 +380,9 @@ func (syncer *Syncer) ValidateTipSet(ctx context.Context, fts *store.FullTipSet)
 
 	for _, b := range fts.Blocks {
 		if err := syncer.ValidateBlock(ctx, b); err != nil {
-			syncer.bad.Add(b.Cid())
+			if !errors.Is(err, ErrTemporal) {
```

Comment:
One nitpick would be to have an `IsPermanent` function that calls this internally. Just to make the logic a little more readable